installers/MacInstaller.hs

```diff
@@ -16,101 +16,76 @@ module MacInstaller
 
 import           Universum
 
-import           Control.Monad (unless, liftM2)
-import           Data.Maybe (fromMaybe)
+import           Control.Monad (unless)
+import           Data.Text (Text)
 import qualified Data.Text as T
 import           System.Directory (copyFile, createDirectoryIfMissing, doesFileExist, renameFile)
-import           System.Environment (lookupEnv)
 import           System.FilePath ((</>), FilePath)
 import           System.FilePath.Glob (glob)
 import           Filesystem.Path.CurrentOS (encodeString)
-import           Turtle (ExitCode (..), echo, proc, procs, which, Managed, with)
+import           Text.Printf (printf)
+import           Turtle (ExitCode (..), echo, proc, procs, shells, which, Managed, with)
 import           Turtle.Line (unsafeTextToLine)
 
 import           RewriteLibs (chain)
 
 import           System.IO (hSetBuffering, BufferMode(NoBuffering))
 
-data InstallerConfig = InstallerConfig {
-    icApi :: String
-  , appNameLowercase :: T.Text
-  , appName :: String
-  , pkg :: T.Text
-  , predownloadChain :: Bool
-  , appRoot :: String
-}
-
--- In both Travis and Buildkite, the environment variable is set to
--- the pull request number if the current job is a pull request build,
--- or "false" if it’s not.
-pullRequestFromEnv :: IO (Maybe String)
-pullRequestFromEnv = liftM2 (<|>) (getPR "BUILDKITE_PULL_REQUEST") (getPR "TRAVIS_PULL_REQUEST")
-  where
-    getPR = fmap interpret . lookupEnv
-    interpret Nothing        = Nothing
-    interpret (Just "false") = Nothing
-    interpret (Just num)     = Just num
+import           Config
+import           Types
```

Comment:
Why was InstallerConfig removed - this was effort from @cleverca22 for mantis installer generalization.

Reply:
of note, the `predownloadChain` section is no longer of use, but we should keep the rest i think

Reply:
The idea is to use Options for that.  I'm discussing this with @cleverca22.